src/docfx/build/page/BuildPage.cs

```diff
@@ -313,6 +312,16 @@ private static (TemplateModel model, JObject metadata) CreateTemplateModel(Conte
             return (model, metadata);
         }
 
+        private static string CreateContent(Context context, HtmlNode html, Document file, HashSet<string> bookmarks = null)
+        {
+            // add bookmark validation
```

Comment:
why do we hide `bookmark validation` within `CreateContent`? If we invoke this explicitly, we do not need to handle null bookmarks.

Reply:
I also think calling `AddBookmarks` explicitly seems better than passing null here